lib/dal/tests/integration_test/frame.rs

```diff
@@ -376,6 +380,99 @@ async fn simple_frames(ctx: &mut DalContext) {
     }
 }
 
+#[test]
+async fn output_sockets_can_have_both(ctx: &mut DalContext) {
```

Comment:
new test! lots more testing to do here to make sure we're handling deletes correctly, nesting, I've got a plan but that can come later